commit 3b74121464fd7626369d93a3f9820f1189bd9374
Author: Jirka Hruška <jirka@fud.cz>
Date:   Wed Sep 1 11:18:43 2010 +0100

    PL_my_cxt_list leaks
    
    [perl #77352]
    
    PL_my_cxt_list was never freed

diff --git a/perl.c b/perl.c
index 94f2b13..8f2a024 100644
--- a/perl.c
+++ b/perl.c
@@ -1151,6 +1151,10 @@ perl_destruct(pTHXx)
 		 (long)cxstack_ix + 1);
     }
 
+    /* the entries in this list are allocated via SV PVX's, so get freed
+     * in sv_clean_all */
+    Safefree(PL_my_cxt_list);
+
     /* Now absolutely destruct everything, somehow or other, loops or no. */
 
     /* the 2 is for PL_fdpid and PL_strtab */

commit c3a99e6d9c4de88ee9b7231b755b7d4f2a3e98a3
Author: Jirka Hruška <jirka@fud.cz>
Date:   Wed Sep 1 10:59:35 2010 +0100

    Memory leak cloning PVGVs
    
    [perl #77352]
    
    In S_sv_dup_common, Perl_rvpv_dup was called twice on a non-GP PVGV value,
    causing the first duped value to be leaked

diff --git a/AUTHORS b/AUTHORS
index 780386b..1a11b90 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -476,6 +476,7 @@ Jim Meyering			<meyering@asic.sc.ti.com>
 Jim Miner			<jfm@winternet.com>
 Jim Richardson
 Jim Schneider			<jschneid@netilla.com>
+Jirka Hruška			<jirka@fud.cz>
 Joachim Huober
 Jochen Wiedmann			<joe@ispsoft.de>
 Jody Belka			<dev-perl@pimb.org>
diff --git a/sv.c b/sv.c
index d2fcb0c..f0287a5 100644
--- a/sv.c
+++ b/sv.c
@@ -10691,6 +10691,7 @@ Perl_sv_dup(pTHX_ const SV *sstr, CLONE_PARAMS* param)
 		else
 		    LvTARG(dstr) = sv_dup_inc(LvTARG(dstr), param);
 	    case SVt_PVGV:
+		/* non-GP case already handled above */
 		if(isGV_with_GP(sstr)) {
 		    GvNAME_HEK(dstr) = hek_dup(GvNAME_HEK(dstr), param);
 		    /* Don't call sv_add_backref here as it's going to be
@@ -10701,8 +10702,7 @@ Perl_sv_dup(pTHX_ const SV *sstr, CLONE_PARAMS* param)
 		    GvSTASH(dstr) = hv_dup(GvSTASH(dstr), param);
 		    GvGP(dstr)	= gp_dup(GvGP(sstr), param);
 		    (void)GpREFCNT_inc(GvGP(dstr));
-		} else
-		    Perl_rvpv_dup(aTHX_ dstr, sstr, param);
+		}
 		break;
 	    case SVt_PVIO:
 		IoIFP(dstr)	= fp_dup(IoIFP(dstr), IoTYPE(dstr), param);
